src/continuum_opacity/opacity_metal.jl

```diff
@@ -0,0 +1,561 @@
+# the only reason that we NEED to load in the electron_configurations data is that it's not
+# completely clear to me exactly how I would compute certain quantities (i.e. the statistical weight and the ground state energy)
+
+include("../Korg.jl")
+using Base
+using Interpolations
+# import Korg.partition_funcs, Korg.atomic_symbols, Korg.numerals, and constants
+
+
+# While the StateID struct and its helper functions may seem a little over-engineered, they
+# primarily exist to document the iSLP and iLV quantities used by TOPBase
+
+"""
+    StateID(iSLP, iLV)
+
+`StateID` represents a given state (electron configuration) of an ion species. The arguments are
+taken from the TOPBase.
+
+# Arguments
+- `iSLP::UInt16`: An integer representing the quantum atomic numbers SLπ (total spin, orbital
+  angular momentum, and parity). The value is given by `100*(2*S+1) + 10*L + π`. Note that TOPBase
+  documentation refers to the parity quantum number as P or π, but for consistency with how TOPBase
+  presents the data, we prefer P in our implementation code.
+- `iLV::UInt8`: Distinguishes between configurations that have the same values of `iSLP`. TOPBase
+  seems to index the configuration based on the energy level of the configuration (with `iLV = 1`
+  corresponding to the lowest energy).
+
+# Notes
+The combination of this struct and an ion species (or neutral element) should be enough to fully
+specify a given electron configuration. For example, consider He I:
+- `iSLP = 100, iLV = 1` has the electron configuration: 1s²
+- `iSLP = 100, iLV = 2` has the electron configuration: 1s 2s
+- `iSLP = 100, iLV = 3` has the electron configuration: 1s 3s
+- `iSLP = 111, iLV = 1` has the electron configuration: 1s 2p
+- `iSLP = 111, iLV = 2` has the electron configuration: 1s 3p
+"""
+struct StateID
+    iSLP::UInt16
+    iLV::UInt8
+    function StateID(iSLP, iLV)
+        @assert 99 < iSLP < 992 # this upper limit is set based on the data source
+        @assert (iSLP % 10) < 2 # trailing digit must be 0 or 1
+        @assert 0 < iLV # the upper limit is set based on the data source
+        new(iSLP, iLV)
+    end
+end
+
+Base.:(<)(a::StateID, b::StateID) = (a.iSLP < b.iSLP) || ((a.iSLP == b.iSLP) && (a.iLV < b.iLV))
+Base.:(==)(a::StateID, b::StateID) = (a.iSLP == b.iSLP) && (a.iLV == b.iLV)
```

Comment:
Because this is an immutable struct, `==` falls back to `===`, which does what you want without a new method.  The same is true for `hash`, I believe.

Reply:
Interesting, then I'll definitely drop `Base.:(==)`.

I'll try to drop my definition of `hash` too (I couldn't find any documentation about the default implementation of `hash`)